pkg/cmd/util/cmd.go

```diff
@@ -31,3 +33,32 @@ func GetDisplayFilename(filename string) string {
 
 	return filename
 }
+
+// ResolveResourceSyntax will return the kind (if found), name and a possible error
+// from the provided arg. If a required kind has been specified, then arg
+// should respect it.
+func ResolveResourceSyntax(mapper meta.RESTMapper, requiredKind, arg string) (string, string, error) {
```

Comment:
I think this would be more obvious and more easily re-used if factored like

```go
// ResolveResource returns the resource, name, and kind of the resourceString
func ResolveResource(resourceString, defaultResource string, mapper meta.RESTMapper) (string, string, string){
}
...
In other code
if _, _, kind := ResolveResource(arg[0], "buildconfigs", mapper); kind != expectedKind{
    // fail
}
```

This factoring would allow the function to parse arguments for functions that require it, determine kinds using the mapper without branches, and make expectations obvious in the code that cares.  It's a little weird to have a `Resolve` function that forces kind matching.